aries_cloudagent/config/default_context.py

```diff
@@ -74,23 +74,24 @@ async def bind_providers(self, context: InjectionContext):
                         # "unpack_message",
                         "get_local_did",
                     ),
-                )
+                ),
+                'wallet.name'
             ),
         )
 
         context.injector.bind_provider(
             BaseLedger,
-            CachedProvider(
-                StatsProvider(
-                    LedgerProvider(),
-                    (
-                        "create_and_send_credential_definition",
-                        "create_and_send_schema",
-                        "get_credential_definition",
-                        "get_schema",
-                    ),
-                )
-            ),
+            #CachedProvider(
```

Comment:
@LukasSMueller  This change results in an error when requesting "POST /credential-definitions" immediately after requesting "POST /schema".
This change seems to support multiple ledgers.
I appreciate for the multi tenant functionality, by the way.

Error log is below.
```
2020-08-04 13:41:38,885 indy.libindy.native.indy.commands.pool INFO 	src/commands/pool.rs:75 | OpenAck handle 14, pool_id 14, result Ok(())
2020-08-04 13:41:38,901 indy.libindy.native.indy.services.ledger INFO 	src/services/ledger/mod.rs:149 | build_get_schema_request() => Ok("{\"reqId\":1596516098900375000,\"identifier\":\"Th7MpTaRZVRYnPiabds81Y\",\"operation\":{\"type\":\"107\",\"dest\":\"Th7MpTaRZVRYnPiabds81Y\",\"data\":{\"name\":\"degree_schema\",\"version\":\"75.21.61\"}},\"protocolVersion\":2}")
2020-08-04 13:41:39,225 indy.libindy.native.indy.services.ledger INFO 	src/services/ledger/mod.rs:143 | build_schema_request() => Ok("{\"reqId\":1596516099225387000,\"identifier\":\"Th7MpTaRZVRYnPiabds81Y\",\"operation\":{\"type\":\"101\",\"data\":{\"name\":\"degree_schema\",\"version\":\"75.21.61\",\"attr_names\":[\"age\",\"name\",\"degree\",\"date\"]}},\"protocolVersion\":2}")
2020-08-04 13:41:39,379 aiohttp.access INFO 127.0.0.1 [04/Aug/2020:04:41:38 +0000] "POST /schemas HTTP/1.1" 200 415 "-" "Go-http-client/1.1"
2020-08-04 13:41:39,384 indy.libindy WARNING _indy_loop_callback: Function returned error 
2020-08-04 13:41:39,385 aries_cloudagent.core.dispatcher ERROR Handler error: AdminServer.make_application.<locals>.activate_wallet
Traceback (most recent call last):
  File "/Users/baegjae/src/aries-cloudagent-python/aries_cloudagent/ledger/indy.py", line 178, in open
    self.pool_handle = await indy.pool.open_pool_ledger(self.pool_name, "{}")
  File "/Users/baegjae/Library/Python/3.8/lib/python/site-packages/indy/pool.py", line 85, in open_pool_ledger
    res = await do_call('indy_open_pool_ledger',
indy.error.PoolLedgerInvalidPoolHandle
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/Users/baegjae/src/aries-cloudagent-python/aries_cloudagent/admin/server.py", line 287, in activate_wallet
    response = await handler(request)
  File "/Users/baegjae/src/aries-cloudagent-python/aries_cloudagent/messaging/credential_definitions/routes.py", line 133, in credential_definitions_send_credential_definition
    async with ledger:
  File "/Users/baegjae/src/aries-cloudagent-python/aries_cloudagent/ledger/indy.py", line 244, in __aenter__
    await self._context_open()
  File "/Users/baegjae/src/aries-cloudagent-python/aries_cloudagent/ledger/indy.py", line 213, in _context_open
    await self.open()
  File "/Users/baegjae/src/aries-cloudagent-python/aries_cloudagent/ledger/indy.py", line 178, in open
    self.pool_handle = await indy.pool.open_pool_ledger(self.pool_name, "{}")
  File "/Users/baegjae/src/aries-cloudagent-python/aries_cloudagent/indy/error.py", line 25, in __exit__
    raise IndyErrorHandler.wrap_error(
  File "/Users/baegjae/src/aries-cloudagent-python/aries_cloudagent/ledger/indy.py", line 178, in open
    self.pool_handle = await indy.pool.open_pool_ledger(self.pool_name, "{}")
  File "/Users/baegjae/Library/Python/3.8/lib/python/site-packages/indy/pool.py", line 85, in open_pool_ledger
    res = await do_call('indy_open_pool_ledger',
aries_cloudagent.ledger.error.LedgerConfigError: Exception when opening pool ledger: Error: Invalid pool handle
  Caused by: Pool with the same name is already opened
```

Reply:
Hi @baegjae, thanks for reporting. I just quickly tested it with postman (creating schema -> creating cred_def) that worked. How are you hitting the endpoints with a script, postman, etc? And how did you start the agent?

Reply:
@LukasSMueller Thanks for reply. I used a script of controller for this. But I can replay error using postman. 1. create schema. 2. get schema from ledger (GET /schemas/{schema_id}) 3. get schema again right after response of 2. Maybe you can see 500 Error at postman and error log at agent.

Reply:
@LukasSMueller One solution I've found is to set ledger.keepalive to 0. It solves those errors. I'm not sure this solution can be final because it could degrade performance.

https://github.com/hyperledger/aries-cloudagent-python/blob/0a034b8f462ca10b7c809e2f5c4995503fe07b5b/aries_cloudagent/ledger/provider.py#L22